DREAM/Optimization/tsgParticleSwarm.hpp

```diff
@@ -80,6 +97,15 @@ class ParticleSwarmState {
     //! \brief Constructor for a particle swarm state with the number of dimensions and the number of particles inferred from
     //! a set of input particle positions \b pp and a set of input particle velocities \b pv.
     ParticleSwarmState(const int num_dimensions, std::vector<double> &&pp, std::vector<double> &&pv);
+    //! \brief Copy constructor.
+    ParticleSwarmState(const ParticleSwarmState &source) = default;
+    //! \brief Move constructor.
+    ParticleSwarmState(ParticleSwarmState &&source) = default;
+    //! \brief Destructor.
+    ~ParticleSwarmState() = default;
```

Comment:
You don't need to specify the default destructor, the compiler will do it automatically. You need the constructors, though.

Reply:
Changed. I removed the destructor.